File: java/L400/LeetCOde0419.java
package L400;

public class LeetCOde0419 {
    public int countBattleships(char[][] board) {
        int ans = 0;
        for (int i = 0;i < board.length;++i) {
            for (int j = 0;j < board[0].length;++j) {
                if (board[i][j] == 'X' && (i - 1 < 0 || board[i - 1][j] == '.')
                   && (j - 1 < 0 || board[i][j - 1] == '.')) {
                       ++ans;
                   } 
            }
        }
        return ans;
    }
}

File: cpp/L400/LeetCode0419.cc
#include <vector>
using std::vector;

class Solution {
public:
    int countBattleships(vector<vector<char>>& board) {
        int ans{0};
        int M = board.size();
        int N = board[0].size();
        for (int i = 0;i < M;++i) {
            for (int j = 0;j < N;++j) {
                if (board[i][j] == 'X') {
                    bool flag = (i - 1 >= 0 && board[i - 1][j] == 'X') || (j - 1 >= 0 && board[i][j - 1] == 'X');
                    if (!flag) {
                        ++ans;
                    }
                }
            }
        }
        return ans;
    }
};